### PR TITLE
Audit reference docs and clean-up

### PR DESCRIPTION
**Pull request by phanimarupaka**
This PR is to audit the overview reference doc. This is displayed when user types `kpt` and `enter`.  There are many global flags which are inherited from `kubectl`.  They should be cleaned as part of separating cli from live library.

### CODE REVIEW

File: site/reference/README.md [hunk 1]
```diff
@@ -43,23 +41,11 @@ fetching package /package-examples/helloworld-set from \
   https://github.com/GoogleContainerTools/kpt to helloworld
```
[the remark this answers is not in the document]
Comment (frankfarzan) [reply site/reference/README.md:41]: https://github.com/GoogleContainerTools/kpt/issues/1645

[the remark this answers is not in the document]
Comment (mikebz) [reply site/reference/README.md:41]: my vote is to go with less.  The book is the overview reference should just have an index to all the command (if that).  Having additional entry points and overviews should be justified by customer demand.

[the remark this answers is not in the document]
Comment (phanimarupaka) [reply site/reference/README.md]: Made changes. Attached site screenshots and cli output after this change.

For cli, a new user should be able to explore and understand kpt cli without needing to read docs. Pasted the output of cli commands below.

File: site/reference/README.md [hunk 2]
```diff
@@ -24,7 +22,7 @@ syntax and pattern of inputs and outputs.
 
 | Command Group | Description                                                                     | Reads From      | Writes To       |
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply site/reference/README.md]: Updated.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**